### PR TITLE
Switch to empty project template

### PR DESCRIPTION
**Pull request by tzerrell**
This PR would change this repo into being an empty project template, rather than a basic example. With this PR, this repo would have a host that runs a no-op guest and then verifies the resulting receipt. This PR also adds `TODO` notes in places where users are most likely to need to add their own code when building a project of their own starting from this repo.

I am looking for feedback on whether this is a change we want to make -- _do_ we want this repository to be a template for beginning users to use for their project, instead of a basic example?

Also, if we do land these changes, I'm thinking we'll want to move the old multiply code into a basic example in the examples repo -- does this plan sound good to others?

### CODE REVIEW

**Comment by pdg744** [comment]
I like this approach -- seems much less likely to be misunderstood than the current starter.

**Comment by tzerrell** [comment]
Based on the feedback here and in https://github.com/risc0/website/pull/71 I think this is the direction we want to go, so I'm marking this as ready for review.

File: project_or_component_name/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    let method_code = std::fs::read(METHOD_NAME_PATH)
+        .expect("Method code should be present at the specified path; did you use the correct *_PATH constant?");
+    let mut prover = Prover::new(&method_code, METHOD_NAME_ID)
+        .expect("Prover should be constructed from valid method source code and corresponding method ID");
+
+    // TODO: Implement communication with the guest here
+
+    // Run prover & generate receipt
+    let receipt = prover.run()
+        .expect("Valid code should be provable if it doesn't overflow the cycle limit. See `embed_methods_with_options` for information on adjusting maximum cycle count.");
```
Comment (pdg744) [inline project_or_component_name/src/main.rs:17]: love this

File: project_or_component_name/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// use risc0_zkvm::serde::{from_slice, to_vec};
+
+fn main() {
+    // Make the prover.
+    let method_code = std::fs::read(METHOD_NAME_PATH)
+        .expect("Method code should be present at the specified path; did you use the correct *_PATH constant?");
+    let mut prover = Prover::new(&method_code, METHOD_NAME_ID)
+        .expect("Prover should be constructed from valid method source code and corresponding method ID");
+
+    // TODO: Implement communication with the guest here
```
Comment (pdg744) [inline project_or_component_name/src/main.rs:13]: Hmm... I'm worried that if I find myself reading this code and am not aware of the associated markdown tutorial you're planning to pair with it, I'll be out at sea

Comment (pdg744) [reply project_or_component_name/src/main.rs:13]: I wonder whether we could provide some more clear signal for how readers should find out how to implement this TODO.

**Review by pdg744 — APPROVED** [review]
Looks good, though I'd like to make sure folks have a clear path toward how to implement the items listed as TODO -- is the intention to land this simultaneous with the markdown tutorial you're working on?

File: project_or_component_name/src/main.rs (hunk 2 above)
Comment (tzerrell) [reply project_or_component_name/src/main.rs:13]: I partially disagree. Certainly the tutorial will help people know what to put here. That said, this is a template, not a tutorial, and I think the TODOs should be brief descriptions of what needs to be added, not long how-tos explaining how to write the code for the section.

Perhaps it makes sense to have a section in the readme which describes where to go to learn more about implementing the various empty sections.

**Review by tzerrell — COMMENTED** [review]
My intention had been to land this before the tutorial lands, but simultaneous with moving the prior multiply example to the examples repo (assuming that also sounds good) -- I don't yet have a PR for that move.

My thinking here is that even without the tutorial, having a template is arguably an improvement on the status quo -- comparing e.g.
```
// TODO: Implement communication with the guest here
```
with
```
// Next we send a & b to the guest
prover.add_input(to_vec(&a).unwrap().as_slice()).unwrap();
prover.add_input(to_vec(&b).unwrap().as_slice()).unwrap();
```
I would argue the former is actually a bit clearer than the latter, given that there are other examples available in the examples repo.

But I could be persuaded otherwise if this continues to seem like a downgrade in the absence of a tutorial -- it's easy enough to wait for the tutorial to be ready.

**Comment by flaub** [comment]
This seems fine, I don't feel like I have a good metric or signal yet on which approach is better. We haven't gotten a lot of feedback yet on what's easier for users to get started with. I'm a little partial to having a working version to start with, but that's perhaps my own personal preference. I'd like to start with a project that I know works and then start doing some tweaks.

**Comment by pdg744** [comment]
OK I agree with your point that this is an improvement over the status quo even without the tutorial

File: project_or_component_name/src/main.rs (hunk 2 above)
Comment (pdg744) [reply project_or_component_name/src/main.rs:13]: Using the README as the bridge to pointers sounds great.

**Comment by tzerrell** [comment]
> This seems fine, I don't feel like I have a good metric or signal yet on which approach is better. We haven't gotten a lot of feedback yet on what's easier for users to get started with. I'm a little partial to having a working version to start with, but that's perhaps my own personal preference. I'd like to start with a project that I know works and then start doing some tweaks.

My current plan is to _move_ the existing multiply example into our examples repo as "factor" (see https://github.com/risc0/risc0-rust-examples/pull/13) in conjunction with this PR. Then people who want to start from a blank template can use this repo, and people who want to start from a working example can start from the examples repo (from "factor" or from one of the more complex examples).

Then in terms of corresponding how-tos, my plan is: 
* A main getting started tutorial that essentially gets people from the template to the factor example (not yet written),
* An explanation of how you would write this template code from scratch (essentially what's in https://github.com/risc0/website/pull/71 but not as the getting started tutorial), and
* Various tutorials/explainers for more complex code, both ones we already have and ones to-be-written.

**Comment by 3lkn** [comment]
* I like the direction this is going and agree with having an empty repo with no example that needs stripping out before making additive changes. In general, I have a strong bias towards separating example from template, but I also like the collection of examples we're gathering and think they'll meet the needs of folks who like working the other way around.
* We chatted about having more Rustlike code comments here, but after review, I don't think I'd change any of the simple code comments. People will be engaging with this as a repo, not a crate, so I don't feel strongly about changing the style.
* As discussed, I'm going to use this as a base for a fresh example to pull in some changes I like from here -- assuming from review feedback above we're close enough to polished that this makes sense.

**Review by 3lkn — APPROVED** [review]

**Comment by tzerrell** [comment]
> * As discussed, I'm going to use this as a base for a fresh example to pull in some changes I like from here -- assuming from review feedback above we're close enough to polished that this makes sense.

This is ready to be used immediately if you want. I am not currently expecting substantial changes to this PR -- in principle we could land it immediately, although I want to finish the website changes first so the website doesn't point people to this repo talking about a multiply example that no longer exists.

**Pull request merged**